### ISSUE TITLE
Repo: main branch doesn't restore coverage from nx cache

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#5832]
### Suggestion

Per @thomasrockhu's _(very helpful! thank you!)_ pointers in https://community.codecov.com/t/incorrect-coverage-changes-reported-base-commit-not-uploading/3899/2:

> ... somewhere in your CI pipeline, you are not uploading coverage reports to Codecov which is why you are seeing the discrepancy

Looking at [logs on a recent `main` branch run](https://github.com/typescript-eslint/typescript-eslint/actions/runs/3254270696/jobs/5342339152):

> Run unit test with coverage for type-utils
> ```plaintext
> Run npx nx test @typescript-eslint/type-utils
> nx run @typescript-eslint/type-utils:test  [remote cache]
> ...
>    Nx read the output from the cache instead of running the command for 1 out of 1 tasks.
> ```

> Run actions/upload-artifact@v3
> ```plaintext
> Warning: No files were found with the provided path: packages/type-utils/coverage/lcov.info. No artifacts will be uploaded.
> ```

I think that's our culprit for incorrect Codecov reports! The `main` branch always restores test results from cache because any PR that gets merged is up-to-date from `main`. 👏 

Solution (I think): adding `coverage/lcov.info` to the `nx test` cache for each package.

This issue replaces #5733.
